acme/acme/client.py

```diff
@@ -306,7 +306,9 @@ def register(self, new_reg: Optional[messages.NewRegistration] = None
 
         """
         new_reg = messages.NewRegistration() if new_reg is None else new_reg
-        response = self._post(self.directory[new_reg], new_reg)
+        with warnings.catch_warnings():
+            warnings.filterwarnings('ignore', '.* non-string keys', DeprecationWarning)
+            response = self._post(self.directory[new_reg], new_reg)
```

Comment:
We'll have to do something like this long term right? If so, I think we should just do it now.
```suggestion
        response = self._post(self.directory['new-reg'], new_reg)
```

Reply:
Longer term we're deleting this code, but fair enough, it looks nicer for sure.

Reply:
Whoops. I failed to notice that this was all in deprecated code.